verifier/build/main.go

```diff
@@ -0,0 +1,345 @@
+		}
+		if st.LatestConsistent.Size > lastHead {
+			klog.V(1).Infof("Found new checkpoint for tree size %d, fetching new leaves", st.LatestConsistent.Size)
+			if err := monitor.From(ctx, lastHead); err != nil {
+				klog.Exitf("monitor.From(%d): %v", lastHead, err)
+			}
+		} else {
+			klog.V(2).Infof("Polling: no new data found; tree size is still %d", st.LatestConsistent.Size)
+		}
+
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			// Go around the loop again.
+		}
+	}
+}
+
+// newReleaseSigStoreFromFlags returns a signature verifier constructed from the
+// flags, or dies trying.
+func newReleaseSigStoreFromFlags() releaseSigStore {
+	osReleaseVerifier1, err := note.NewVerifier(*osReleasePubKey1)
+	if err != nil {
+		klog.Exitf("Failed to construct OS release verifier: %v", err)
+	}
+	osReleaseVerifier2, err := note.NewVerifier(*osReleasePubKey2)
+	if err != nil {
+		klog.Exitf("Failed to construct OS release verifier: %v", err)
+	}
+	appletReleaseVerifier, err := note.NewVerifier(*appletReleasePubKey)
+	if err != nil {
+		klog.Exitf("Failed to construct applet release verifier: %v", err)
+	}
+	releaseVerifiers := note.VerifierList(osReleaseVerifier1, osReleaseVerifier2, appletReleaseVerifier)
+
+	dir, err := os.MkdirTemp("", "armored-witness-build-keys")
+	if err != nil {
+		klog.Exitf("Failed to create temp directory: %v", err)
+	}
+	logFile := filepath.Join(dir, "pubkey_log.pub")
+	os1File := filepath.Join(dir, "pubkey_os1.pub")
+	os2File := filepath.Join(dir, "pubkey_os2.pub")
+	appletFile := filepath.Join(dir, "pubkey_applet.pub")
+	if err := os.WriteFile(logFile, []byte(*logPubKey), 0644); err != nil {
+		klog.Exitf("Failed to create public key file: %v", err)
+	}
+	if err := os.WriteFile(os1File, []byte(*osReleasePubKey1), 0644); err != nil {
+		klog.Exitf("Failed to create public key file: %v", err)
+	}
+	if err := os.WriteFile(os2File, []byte(*osReleasePubKey2), 0644); err != nil {
+		klog.Exitf("Failed to create public key file: %v", err)
+	}
+	if err := os.WriteFile(appletFile, []byte(*appletReleasePubKey), 0644); err != nil {
+		klog.Exitf("Failed to create public key file: %v", err)
+	}
+
+	return releaseSigStore{
+		osV1: osReleaseVerifier1,
+		osV2: osReleaseVerifier2,
+		appV: appletReleaseVerifier,
+		allV: releaseVerifiers,
+		envs: []string{
+			fmt.Sprintf("REST_DISTRIBUTOR_BASE_URL=%s", *distributorURL),
+			fmt.Sprintf("FT_LOG_URL=%s", *logURL),
+			fmt.Sprintf("FT_BIN_URL=%s", *binURL),
+			fmt.Sprintf("LOG_ORIGIN=%s", *logOrigin),
+			fmt.Sprintf("LOG_PUBLIC_KEY=%s", logFile),
+			fmt.Sprintf("APPLET_PUBLIC_KEY=%s", appletFile),
+			fmt.Sprintf("OS_PUBLIC_KEY1=%s", os1File),
+			fmt.Sprintf("OS_PUBLIC_KEY2=%s", os2File),
+		},
+		cleanup: func() {
+			os.RemoveAll(dir)
+		},
+	}
+}
+
+// releaseSigStore stores the signatures used for verifying releases.
+// Having such an object avoids needing to pass lots of state as args, or parsing
+// keys in different places.
+type releaseSigStore struct {
```

Comment:
I had a triple-take when I read this :)
How about `releaseVerifiers`?

Reply:
Picked another name, but glad I got a reaction out of you :-)